application/ui/components/param-row.jsx

```diff
@@ -0,0 +1,221 @@
+                            <ArrayObject
+                                key={component.props.key}
+                                param={component.props.param.param}
+                                onChange={component.props.onChange}
+                            />
+                        </td>
+                    </tr>
+                );
+                break;
+            case 'custom-object':
+                rowParam = (
+                    <tr>
+                        <td colSpan={5}>
+                            <CustomObject name='' onChange={component.props.onChange} />
+                        </td>
+                    </tr>
+                );
+                break;
+
+            // inline params
+            case 'enum':
+                var enumValues = _.map(component.props.param.enumValues, function (value) {
+                    return {
+                        label : value,
+                        value : value
+                    };
+                });
+                inlineParam = (
+                    <Select value={component.props.value} key={component.props.key} options={enumValues} onChange={component.props.onChange} />
+                );
+                break;
+            case 'boolean':
+                inlineParam = (
+                    <Select value={component.props.value} key={component.props.key} options={[{value: false, label: 'false'}, {value: true, label: 'true'}]} onChange={component.props.onChange} />
+                );
+                break;
+            case 'string':
+            case 'number':
+                inlineParam = (
+                    <Text key={component.props.key} type={component.props.param.type} onChange={component.props.onChange} />
+                );
+                break;
+            case 'stripe-token':
+                inlineParam = (
+                    <StripeToken paramName={component.props.param.name} onChange={component.props.onChange} />
+                );
+                break;
+        }
+
+        if (component.props.simple === true) {
+            return (
+                <tbody>
+                    <tr>
+                        <td><code>{component.props.param.name}</code></td>
+                        <td><code>{component.props.param.type}</code></td>
+                        <td>{inlineParam}</td>
+                    </tr>
+                    {rowParam}
+                </tbody>
+            );
+        }
+        description = component.renderDescription(component.props.param);
+
+        if (description) {
+            description = (
+                <tr>{description}</tr>
+            );
+        }
+
+        return (
+            <tbody>
+                <tr>
+                    <td><code>{component.props.param.name}</code></td>
+                    <td>{includeCheckboxComponent}</td>
+                    <td>{nullCheckboxComponent}</td>
+                    <td><code>{component.props.param.type}</code></td>
```

Comment:
We could get rid of the `component` alias in this method and refer to `this` instead. There aren't any callbacks that require it.